test/parallel/xyk-pallet.API.burnLiquidity.test.ts

```diff
@@ -241,9 +246,113 @@ describe("xyk-pallet - Burn liquidity tests: when burning liquidity you can", ()
       burnAmount
     );
   });
+
+  test("Given a pool with 2 users with activated rewards WHEN more than one period last AND the user burn all liquidity THEN pool is destroyed but users can still claim pending rewards", async () => {
+    const testUser2 = new User(keyring);
+    keyring.addPair(testUser2.keyRingPair);
+    [firstCurrency] = await Assets.setupUserWithCurrencies(
+      testUser1,
+      [new BN(defaultCurrecyValue)],
+      sudo
+    );
+
+    testUser1.addAsset(MGA_ASSET_ID);
+    testUser2.addAsset(MGA_ASSET_ID);
+
+    await Sudo.batchAsSudoFinalized(
+      Assets.FinalizeTge(),
+      Assets.initIssuance(),
+      Assets.mintNative(testUser1),
+      Assets.mintToken(firstCurrency, testUser2, new BN(defaultCurrecyValue)),
+      Assets.mintNative(testUser2)
+    );
+
+    const mga = await getMangataInstance();
+
+    await mga.createPool(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID.toString(),
+      new BN(defaultCurrecyValue).div(new BN(2)),
+      firstCurrency.toString(),
+      new BN(defaultCurrecyValue).div(new BN(2))
+    );
+
+    const liquidityAssetId = await getLiquidityAssetId(
+      MGA_ASSET_ID,
+      firstCurrency
+    );
+
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAs(
+        testUser2,
+        Xyk.mintLiquidity(
+          MGA_ASSET_ID,
+          firstCurrency,
+          new BN(defaultCurrecyValue).div(new BN(2))
+        )
+      ),
+      Assets.promotePool(liquidityAssetId.toNumber(), 20)
+    );
+    await activateLiquidity(
+      testUser1.keyRingPair,
+      liquidityAssetId,
+      new BN(defaultCurrecyValue).div(new BN(2))
+    );
+    await activateLiquidity(
+      testUser2.keyRingPair,
+      liquidityAssetId,
+      new BN(defaultCurrecyValue).div(new BN(2))
+    );
+
+    await waitForRewards(testUser1, liquidityAssetId);
+
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAs(
+        testUser1,
+        Xyk.burnLiquidity(
+          MGA_ASSET_ID,
+          firstCurrency,
+          new BN(defaultCurrecyValue).div(new BN(2))
+        )
+      ),
+      Sudo.sudoAs(
+        testUser2,
+        Xyk.burnLiquidity(
+          MGA_ASSET_ID,
+          firstCurrency,
+          new BN(defaultCurrecyValue).div(new BN(2))
+        )
+      )
+    );
+
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+    await testUser2.refreshAmounts(AssetWallet.AFTER);
+
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liquidityAssetId)),
+      Sudo.sudoAs(testUser2, Xyk.claimRewardsAll(liquidityAssetId))
+    );
```

Comment:
@AlexChetverov  can you please check that the pool is empty or destroyed? It should have 0,0 when retrieving pool balance

Reply:
Added